### PR TITLE
[SHIPIT-0513] fully working client implementation

### PR DESCRIPTION
**Pull request by kaisershahid**
set of fixes that allow unleash client to fully work with server

### CODE REVIEW

File: src/Metrics.php [hunk 1]
```diff
@@ -65,18 +76,22 @@ public function init()
         if ($this->metricsInterval > 0) {
             $this->registerInstance();
             $this->sendMetrics();
```
Comment (bassrock) [inline src/Metrics.php]: I wonder if we need this extra send metrics call since we have the register shutdown one.

File: src/Repository.php [hunk 2]
```diff
@@ -50,25 +60,24 @@ public function __construct(
...
+        );
+        $this->storage->addListener(
+            'ready',
+            function () {
+                $this->dispatch('ready');
+            }
+        );
     }
 
     public function timedFetch()
```
Comment (bassrock) [inline src/Repository.php]: Do we even need this method?

**Review by bassrock — APPROVED** [review]
Just had a couple comments but they are nowhere near blocking

File: src/Metrics.php (hunk 1 above)
Comment (kaisershahid) [reply src/Metrics.php]: it's redundant, will remove

File: src/Repository.php (hunk 2 above)
Comment (kaisershahid) [reply src/Repository.php]: nope, will remove

**Pull request merged**